tensordict/nn/storage.py

```diff
@@ -280,13 +284,49 @@ class SipHash(torch.nn.Module):
 
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         hash_values = []
-        for x_i in x.detach().cpu().view(torch.uint8).numpy():
+        for x_i in x.detach().cpu().numpy():
             hash_value = hash(x_i.tobytes())
             hash_values.append(hash_value)
 
         return torch.tensor(hash_values, dtype=torch.int64)
 
 
+class RandomProjectionHash(SipHash):
+    """A module that combines random projections with SipHash to get a low-dimensional tensor, easier to embed through SipHash.
+
+    This module requires sklearn to be installed.
+
+    """
+
+    def __init__(
+        self,
+        n_components=16,
+        projection_type: str = "gaussian",
+        dtype_cast=torch.float16,
+        **kwargs,
+    ):
+        super().__init__()
+        from sklearn.random_projection import (
+            GaussianRandomProjection,
```

Comment:
Is this the same as MLP where the weight are initialized from gaussian distribution?

Reply:
No I believe there is some fitting to the data involved, where it's trying to find the best separation for your examples using the projection.

Reply:
I did a quick check and GaussianRandomProjection does not seem to use X in the fit function, it only uses it to get input shape:
https://github.com/scikit-learn/scikit-learn/blob/70fdc843a4b8182d97a3508c1a426acc5e87e980/sklearn/random_projection.py#L364